source/ncengine/graphics2/diligent/pass/PassManifest.cpp

```diff
@@ -80,15 +82,32 @@ void PassManifest::RegisterPass(PassDesc desc)
         throw nc::NcError("The pass was already registered");
     }
 
+    if (desc.colorSink != NoTarget && desc.colorSink != SwapChainColorRTIndex)
```

Comment:
All the ifs here do the same checks. I think it'd be a lot less noisy if they were stored in local bools and/or the count incrementing in the switch cases was extracted to a function.